### PR TITLE
Stargaze Fair Burn v1.1.0

### PR DESCRIPTION
**Pull request by tasiov**
# Fair Burn Changes

This PR implements a `fee_manager` address in the fair burn contract config. This address will receive the protocol's share of fees for denoms other than STARS.

### CODE REVIEW

File: contracts/fair-burn/src/migrate.rs [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        Version::parse(&prev_contract_version.version).unwrap()
+            < Version::parse(CONTRACT_VERSION).unwrap(),
+        StdError::generic_err("Must upgrade contract version")
+    );
+
+    let config_v1_0 = CONFIG_V1_0.load(deps.storage)?;
+    let config = Config {
+        fee_percent: config_v1_0.fee_percent,
+        fee_manager: deps.api.addr_validate(&msg.fee_manager)?,
+    };
```
Comment (jhernandezb) [inline contracts/fair-burn/src/migrate.rs:51]: Nice 👍

File: contracts/fair-burn/src/execute.rs [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+                    if let Some(dist_coin) = dist_coin {
+                        recipient_funds.push(dist_coin);
+                    }
+                } else {
+                    let fee_manager_coin = coin(
+                        protocol_coin.amount.u128() + dist_coin.map_or(0u128, |c| c.amount.u128()),
+                        protocol_coin.denom,
+                    );
+                    fee_manager_funds.push(fee_manager_coin);
+                }
```
Comment (jhernandezb) [inline contracts/fair-burn/src/execute.rs:83]: Ok this was a bit confusing maybe should add just a few doc comments
recipient is if there is a dev address to share fairburn right ?

**Review by jhernandezb — APPROVED** [review]

File: contracts/fair-burn/src/execute.rs (hunk 2 above)
Comment (tasiov) [reply contracts/fair-burn/src/execute.rs:83]: Yes that's right, typically won't be used for our stuff

**Review by MightOfOaks — APPROVED** [review]

File: contracts/fair-burn/src/execute.rs [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+            }
+        }
+    }
+
+    if !fee_manager_funds.is_empty() {
+        response = response.add_message(BankMsg::Send {
```
Comment (shanev) [inline contracts/fair-burn/src/execute.rs:89]: There's a helper in `sg-std` now for `BankMsg::Send` if you want to use it.

**Review by shanev — APPROVED** [review]

**Pull request merged**